### ISSUE TITLE
remove replicasSynced?

### ISSUE THREAD

**Issue by joel-bluedata** [bluek8s/kubedirector#82]
Currently we don't process create-pending-state or deleting-state members until the relevant statefulset has achieved the desired number of replicas. This means that none of the members in that role can change state until that point.

For create-pending-state this is probably good; it somewhat helps batch up the ready-member-notifies since we don't start configuring any of the new members until all the pods exist.

For deleting-state this is not as helpful? We've already done the notifies. It would be slightly nicer if the deleting-state members could actually disappear one by one as the related pods go away.
